challenges/2022-04-26-dojo-anniversary-edition/solutions/javascript/matthewfee/Level4.js

```diff
@@ -0,0 +1,303 @@
+class Block {
+  constructor(startPosition) {
+    this.startPosition = startPosition
+    this.gridPosition = startPosition
+    this.isDownVertically = null
+    this.isDownHorizontally = null
+    this.isDown = null
+    this.moveHistory = []
+    this.destination = null
+    this.xDistance = null
+    this.yDistance = null
+    this.updateOrientation()
+  }
+
```

Comment:
I really like how you've abstracted out movements.

Can something be this.isDown and !this.isDown at the same time?

If not this should perhaps be an if / else